v2/internal/runner/runner.go

```diff
@@ -162,31 +197,60 @@ func (r *Runner) RunEnumeration() {
 		gologger.Fatalf("Error, no templates found in directory: '%s'\n", r.options.Templates)
 	}
 
-	var results bool
+	// precompute request count
+	var totalRequests int64 = 0
+	var totalTemplates int = len(matches)
 	for _, match := range matches {
 		t, err := r.parse(match)
 		switch t.(type) {
 		case *templates.Template:
 			template := t.(*templates.Template)
+			totalRequests += template.GetHTTPRequestsCount()
+		default:
+			gologger.Errorf("Could not parse file '%s': %s\n", r.options.Templates, err)
+		}
+	}
+
+	// track progress
+	p := progress.NewProgress(nil)
+	p.SetupGlobalProgressbar(r.inputCount, len(matches), r.inputCount * totalRequests)
+
+	var results bool
+	for i, match := range matches {
+		t, err := r.parse(match)
+		switch t.(type) {
+		case *templates.Template:
+			template := t.(*templates.Template)
+
+			p.SetupTemplateProgressbar(i, totalTemplates, template.ID, r.inputCount * template.GetHTTPRequestsCount())
+
 			for _, request := range template.RequestsDNS {
-				dnsResults := r.processTemplateRequest(template, request)
+				dnsResults := r.processTemplateWithList(p, template, request)
 				if dnsResults {
 					results = dnsResults
 				}
 			}
+
 			for _, request := range template.RequestsHTTP {
-				httpResults := r.processTemplateRequest(template, request)
+				httpResults := r.processTemplateWithList(p, template, request)
 				if httpResults {
 					results = httpResults
 				}
 			}
+
 		case *workflows.Workflow:
 			workflow := t.(*workflows.Workflow)
 			r.ProcessWorkflowWithList(workflow)
 		default:
+			p.StartStdCapture()
```

Comment:
Instead of littering with start/stop capture at every fmt.print/gologger.Log instance, we could have the gologger accept custom `io.Writer` instances either for stdout, stderr or both.